src/Mirza/SupplyChain/EventUtils.hs

```diff
@@ -135,19 +141,29 @@ getAction (TransactWhat (TransactionDWhat act _ _ _)) = Just act
 getAction (AggWhat (AggregationDWhat act _ _))        = Just act
 
 
-findInstLabelId :: InstanceLabelEPC -> DB context err [PrimaryKeyType]
+findInstLabelId :: InstanceLabelEPC -> DB context err (Maybe PrimaryKeyType)
 findInstLabelId (GIAI cp sn) = findInstLabelId' cp sn Nothing Nothing Nothing
 findInstLabelId (SSCC cp sn) = findInstLabelId' cp sn Nothing Nothing Nothing
 findInstLabelId (SGTIN cp msfv ir sn) = findInstLabelId' cp sn msfv (Just ir) Nothing
 findInstLabelId (GRAI cp at sn) = findInstLabelId' cp sn Nothing Nothing (Just at)
 
 
+findInstLabelIdByUrn :: InstanceLabelEPC -> DB context err [PrimaryKeyType]
```

Comment:
This function isn't used (yet)?

Reply:
That's correct. It was there more so as a proof of concept, but I think we can start using it (now that we have a unique field `label_urn`).